lib/user.rb

```diff
@@ -0,0 +1,32 @@
+require 'bcrypt'
+require_relative './database_connection'
+
+class User
+  def self.create(name:, username:, email:, password:)
+    encrypted_password = BCrypt::Password.create(password)
+
+    result = DatabaseConnection.query("INSERT INTO users \
+      (name, username, email, password) \
+      VALUES('#{name}', '#{username}', '#{email}', '#{encrypted_password}') \
+      RETURNING id, name, username, email;")
+    User.new(id: result[0]['id'], name: result[0]['name'],\
+      username: result[0]['username'], email: result[0]['email'])
```

Comment:
You can remove the multiple calls to `[0]` if you call `.first` on the result set returned from the database.